sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,423 @@
+            The predicted probabilities for X.
+        state_info : List or None
+            A List containing the state info for each decision in X. contains
+            information for future decisions on the data. Inputs should be None for the
+            first decision made, the returned List new_state_info for subsequent
+            decisions.
+
+        Returns
+        -------
+        decisions : List
+            A List of booleans, containing the decision of whether a prediction is safe
+            to use or not.
+        new_state_info : List
+            A List containing the state info for each decision in X, contains
+            information for future decisions on the data.
+        """
+        X = check_X(X, coerce_to_numpy=True)
+
+        n_instances, _, series_length = X.shape
+        idx = self._classification_point_dictionary.get(series_length, -1)
+
+        if idx == -1:
+            raise ValueError(
+                f"Input series length does not match the classification points produced"
+                f" in fit. Current classification points: {self._classification_points}"
+            )
+
+        # If this is the smallest dataset, there should be no state_info, else we
+        # should have state info for each, and they should all be the same length
+        if state_info is None and (
+            idx == 0 or idx == len(self._classification_points) - 1
+        ):
+            state_info = [(0, 0, 0) for _ in range(n_instances)]
+        elif isinstance(state_info, list) and idx > 0:
+            if not all(si[0] == idx for si in state_info):
+                raise ValueError("All input instances must be of the same length.")
+        else:
+            raise ValueError(
+                "state_info should be None for first time input, and a list of "
+                "state_info outputs from the previous decision making for later inputs."
+            )
+
+        # if we have the full series, always return true
+        if idx == len(self._classification_points) - 1:
+            return [True for _ in range(n_instances)], None
+
+        # find predicted class for each instance
+        rng = check_random_state(self.random_state)
+        preds = [
+            int(rng.choice(np.flatnonzero(prob == prob.max())))
+            for prob in X_probabilities
+        ]
+
+        # make a decision based on the one class svm prediction
+        if self._svms[idx] is not None:
+            X_svm = np.hstack((X_probabilities, np.ones((len(X), 1))))
+
+            for i in range(len(X)):
+                for n in range(self.n_classes_):
+                    if n != preds[i]:
+                        X_svm[i][self.n_classes_] = min(
+                            X_svm[i][self.n_classes_], X_svm[i][preds[i]] - X_svm[i][n]
+                        )
+
+            decisions = self._svms[idx].predict(X_svm) == 1
+        else:
+            decisions = [False for _ in range(n_instances)]
```

Comment:
Same: Maybe initialize decisions to False instead of branching

Reply:
Same as above.

Reply:
Same as above  ;)